elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/hibernate/hql/RootCollectionFetchQueryBuilder.java

```diff
@@ -49,7 +49,8 @@ public Query build() {
 
             //Build the JOIN clause
             String joinClause =  getJoinClauseFromFilters(filterExpression.get())
-                    + extractToOneMergeJoins(entityClass, entityAlias);
+                    + extractToOneMergeJoins(entityClass, entityAlias)
+                    + explicitSortJoins(sorting, entityClass);
```

Comment:
You will need to make similar changes to SubCollectionFetchQueryBuilder.java

Also, I recommend adding a test on a subcollection: /foo/123/bar where your new sorting join logic can be validated (sort on bar.blah.someField or something).